src/app/Library/CrudPanel/Traits/Support/HasMacros.php

```diff
@@ -0,0 +1,18 @@
+<?php
```

Comment:
This code is added to the `CrudField`, `CrudColumn` objects that don't use the backpack `Macroable` trait, but uses the `Laravel/Macroable` trait. 

You cannot extend a trait to add that functionality, so you have two options:
- duplicate the code in both `CrudField` and `CrudColumn`;
- create a trait to share the functionality in both of them. 

I went for the second, let me know if I am missing something.

Reply:
Ooooh. `CrudColumn` and `CrudField` use `Laravel/Macroable`... why? Why don't they use our `Macroable`? From what I know the only difference is that ours throws a visible error if something can't be macroed... right?

Reply:
This is the only thing missing to do here.

Our macro trait is not compatible with the `CrudField` and `CrudColumn` implementations. 

The overridden function you talked about is the blocker. `macro()`.
It works fine in `CrudPanel` because there is no parameters in the constructor. In both `CrudField` and `CrudColumn` the `$name` is a mandatory parameter, that we cannot properly resolve in the macro register process. 
Calling `new static()` in that method would obviously fail when there are mandatory parameters.

Did I make the problem clear enough ? Do you see any other way to deal with this limitation ? 

ping @tabacitu